vms/evm/database/wrapped_database.go

```diff
@@ -0,0 +1,61 @@
+// Copyright (C) 2019-2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package database
+
+import (
+	"errors"
+
+	"github.com/ava-labs/libevm/ethdb"
+
+	"github.com/ava-labs/avalanchego/database"
+)
+
+var (
+	_ ethdb.Batch         = (*ethBatchWrapper)(nil)
+	_ ethdb.KeyValueStore = (*ethDBWrapper)(nil)
+
+	ErrSnapshotNotSupported = errors.New("snapshot is not supported")
+	ErrStatNotSupported     = errors.New("stat is not supported")
+)
+
+func WrapDatabase(db database.Database) ethdb.KeyValueStore { return ethDBWrapper{db} }
+
+type ethDBWrapper struct{ database.Database }
+
+func (ethDBWrapper) Stat(string) (string, error) { return "", ErrStatNotSupported }
+
+func (db ethDBWrapper) NewBatch() ethdb.Batch { return ethBatchWrapper{db.Database.NewBatch()} }
+
+// TODO: propagate size through avalanchego Database interface
+func (db ethDBWrapper) NewBatchWithSize(int) ethdb.Batch {
+	return ethBatchWrapper{db.Database.NewBatch()}
```

Comment:
We could just return `db.NewBatch()` here

Reply:
If we do that, then we does not implement `ethdb.KeyValueStore`:
```
		have NewBatch() "github.com/ava-labs/avalanchego/database".Batch
		want NewBatch() ethdb.Batch
```